clients/c/src/orpProtocol.c

```diff
@@ -1109,8 +1112,17 @@ static bool orp_ProtocolEncode_v1
         {
             break;
         }
-        packet[ORP_OFFSET_SEQ_NUM]     = (msg->sequenceNum & 0x00FF);
-        packet[ORP_OFFSET_SEQ_NUM + 1] = (msg->sequenceNum & 0xFF00) >> 8;
+
+        // Copy Sequence Number from Request or increment it in case of request sent by the device
+        if ((packet[ORP_OFFSET_PACKET_TYPE] != ORP_PKT_SYNC_SYN) && isupper(packet[ORP_OFFSET_PACKET_TYPE]))
+        {
+            // The device will send a request -> increment the Sequence Number
+            last_received_seq_number += 1;
```

Comment:
So this variable does not contain the "last_received_seq_number" anymore, but the "seq_number" of the next packet. I guess that it would be easier to understand using a new variable.